iceprog/iceprog.c

```diff
@@ -438,43 +553,43 @@ int main(int argc, char **argv)
 	if (devstr != NULL) {
 		if (ftdi_usb_open_string(&ftdic, devstr)) {
 			fprintf(stderr, "Can't find iCE FTDI USB device (device string %s).\n", devstr);
-			error();
+			error(2);
```

Comment:
I agree that this isn't nice; maybe even an old version of the file has been truncated. The correct way of handling this would be writing to a temporary file whose name is based on the target file (to make sure it's on the same file system) and on success renaming it, replacing the target file. It believe this may be a bit overkill here, but if you think it's worth the additional complexity, I'll implement it.